A80-tcp-telemetry.md

```diff
@@ -0,0 +1,96 @@
+# A80: Export TCP Telemetry from gRPC
+
+**Author(s)**: Aananth V (@aananthv), Nana Pang (@nanahpang), Yash Tibrewal (@yashykt), Yousuk Seung  (@yousukseung)
+**Approver(s)**: Craig Tiller (@ctiller), Mark Roth (@markdroth)  
+**Status**: In Review  
+**Implemented in**: C-Core  
+**Last updated**: Oct 21, 2025
+
+## Abstract
+
+This document proposes collecting and exposing new TCP Endpoint Level Telemetry to gRPC for improved network analysis and debugging.
+
+## Background
+
+The Linux Kernel exposes two telemetry hooks that can be used to collect TCP-level metrics.
+
+1. **TCP socket state** can be retrieved using the `getsockopt()` system call with `level` set to `IPPROTO_TCP` and `optname` set to `TCP_INFO`. The state is returned in a `struct tcp_info` which gives details about the TCP connection. At present, the machinery to collect such information is available on Linux 2.6 or later kernels.  
+2. **Per-message transmission timestamps** can be collected from TCP sockets using the [SO\_TIMESTAMPING](https://docs.kernel.org/networking/timestamping.html) interface. At present, this is available on Linux 2.6 or later kernels. These timestamps can be very valuable for diagnosing network level issues and can be used to break down the time spent in the "network".
+
+\[[A79](https://github.com/grpc/proposal/blob/master/A79-non-per-call-metrics-architecture.md)\] provides a framework for adding non-per-call metrics in gRPC. This document uses that framework to expose the proposed TCP Latency metrics.
+
+### *Related Proposals:*
+
+\*   \[[A79](https://github.com/grpc/proposal/blob/master/A79-non-per-call-metrics-architecture.md)\]: gRPC Non-Per-Call Metrics Framework
+
+## Proposal
+
+This document proposes exporting the following TCP metrics from gRPC to improve the network debugging capabilities for gRPC users.
```

Comment:
Just to avoid future questions, let's explicitly state that these metrics will be implemented inside the gRPC endpoint layer.